### PR TITLE
framework/task_manager: Modify late unregister with pthread

### PR DESCRIPTION
**Pull request by jeongchanKim**
1.     framework/task_manager: Modify late unregister with pthread instead of lpwork
    
    If there is no lpwork, task manager cannot support late unregister.
    So changed to pthread which treats the late unregister functionality.

2.     testcase/task_manager : Modify tc for removing unnecessary semaphore and add sleep

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [aca689a8c6997bc039108123f4f152193875b1f4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [aca689a8c6997bc039108123f4f152193875b1f4] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -268,15 +268,26 @@ static void taskmgr_execute_unregister(int handle)
 	handle_cnt--;
 }
 
-static void taskmgr_late_unregister(void *arg)
+static void *taskmgr_late_unregister(void *arg)
 {
+	int ret;
+	while (1) {
+		ret = taskmgr_get_task_state((int)arg);
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_core.c]: Is ```arg``` right? not ```*arg```?

File: framework/src/task_manager/task_manager_core.c [hunk 2]
```diff
@@ -268,15 +268,26 @@ static void taskmgr_execute_unregister(int handle)
...
+		if (ret == TM_APP_STATE_STOP || ret == TM_APP_STATE_UNREGISTERED) {
+			break;
+		}
+		sleep(TM_INTERVAL_TRY_UNREGISTER);
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager_core.c]: done.

**Comment by seinfra** [comment]
Target : [cd0520fdb1fff2289ed14ffdc0eec4979d162fbf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cd0520fdb1fff2289ed14ffdc0eec4979d162fbf] - Code Rule Check OK.

File: framework/src/task_manager/task_manager_core.c [hunk 3]
```diff
@@ -247,40 +249,61 @@ static void taskmgr_execute_unregister(int handle)
...
+	if (IS_INVALID_HANDLE(unregister_handle)) {
 		return TM_INVALID_PARAM;
 	}
 
-	state = taskmgr_get_task_state(handle);
+	state = taskmgr_get_task_state(unregister_handle);
 	if (state == TM_APP_STATE_UNREGISTERED) {
 		return TM_UNREGISTERED_APP;
 	}
 #if defined(CONFIG_SCHED_WORKQUEUE) || defined(CONFIG_LIB_USRWORK)
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_core.c]: You need to change this conditional and including header (line 41).

File: framework/src/task_manager/task_manager_core.c (hunk 1 above)
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager_core.c]: arg is correct.

File: framework/src/task_manager/task_manager_core.c [hunk 4]
```diff
@@ -268,15 +268,26 @@ static void taskmgr_execute_unregister(int handle)
...
 	taskmgr_execute_unregister((int)arg);
+	return NULL;
 }
 
 static int taskmgr_unregister(int handle)
 {
 	int state;
-	struct work_s *wq;
+	int ret;
+	pthread_t pth;
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager_core.c]: done.

**Pull request merged**